packages/router/types.ts

```diff
@@ -5,13 +5,22 @@ interface Router<T = null> extends Package {
     frontity?: Package["state"]["frontity"];
     router: {
       link: string;
+      method: SetOptions["method"];
+      state: SetOptions["state"];
     };
   };
   actions: {
     router: {
-      set: Action<T extends null ? Router : T, string>;
+      set:
+        | Action<T extends null ? Router : T, string>
+        | Action<T extends null ? Router : T, string, SetOptions>;
     };
   };
 }
 
 export default Router;
+
+export interface SetOptions {
+  method?: "push" | "replace" | "pop";
```

Comment:
I would prefer not to expose `"pop"` to the users because they will see it in TS autocompletion and it is meant to be used internally only.

I didn't like the `isPopState` flag either but... I don't know. Maybe doing manual casting of that type internally? I can't think of a nice way to do it right now, honestly.